crossterm_input/examples/async_input.rs

```diff
@@ -14,27 +21,126 @@ pub fn read_async_until() {
     // for more information checkout `crossterm_screen` or  use crossterm with the `screen` feature flag.
 
     // init some modules we use for this demo
-    let input = input();
+    let screen = Screen::new(true);
+    let input = TerminalInput::from_output(&screen.stdout);
 
-    let mut stdin = input.read_until_async(b'\r').bytes();
+    input.enable_mouse_mode().unwrap();
 
+    let mut stdin = input.read_async().bytes();
     for _i in 0..100 {
+    //     let mut buf = Vec::new();
+    //     let mut switch = true;
+    //     while switch {
+    //         let a = stdin.next();
+    //         if a.is_none() {
+    //             thread::sleep(time::Duration::from_millis(100));
+    //             switch = false;
+    //         } else {
+    //             buf.push(a.unwrap().unwrap() as char);
+    //         }
+    //     }
+    //     println!("{:?}", buf);
+    // }
         let a = stdin.next();
-
-        println!("pressed key: {:?}", a);
-
-        if let Some(Ok(b'\r')) = a {
-            println!("The enter key is hit and program is not listening to input anymore.");
-            break;
-        }
-
-        if let Some(Ok(b'x')) = a {
-            println!("The key: x was pressed and program is terminated.");
-            break;
-        }
-
+        if a.is_none() {
+            thread::sleep(time::Duration::from_millis(100));
+            continue
+        } else {
+            let event = parse_event(a.unwrap().unwrap(), &mut stdin);
```

Comment:
This seems really cool, lot's of unwraps though.  Did you test this on windows too?